daemon/repo-mgr.c

```diff
@@ -2122,6 +2122,8 @@ add_path_to_index (SeafRepo *repo, struct index_state *istate,
     full_path = g_build_filename (repo->worktree, path, NULL);
 
     if (seaf_stat (full_path, &st) < 0) {
+        send_file_sync_error_notification (repo->id, repo->name, path,
+                                           SYNC_ERROR_ID_INDEX_ERROR);
```

Comment:
You should also send error notification when add_to_index() function returns -1. Check every place that calls this function and make changes.